engine/src/main/java/org/terasology/world/block/BlockComponent.java

```diff
@@ -12,9 +12,9 @@
  */
 public final class BlockComponent implements Component {
     @Replicate
-    Vector3i position = new Vector3i();
+    protected Vector3i position = new Vector3i();
     @Replicate
-    public Block block;
+    public Block block; // use getter
```

Comment:
This currently has 69 usages in Omega, with two write accesses, one not coming from this class:

![image](https://user-images.githubusercontent.com/1448874/109554854-168e9d80-7ad5-11eb-9062-5a3d24405de9.png)

How are we planning to do these follow ups?

Before merging this we should get the write access sorted out so that we can eventually set this field non-public.

Reply:
```suggestion
    public Block block;
```

Reply:
We could potentially replace the write access with the typical thing you do with immutable types: create a new instance and use that instead:

In `EntityAwareWorldProvider.java`
```java
BlockComponent newBlockComponent = new BlockComponent(type, oldBlockComponent.getPosition());
blockEntity.saveComponent(newBlockComponent);
```